server/plugin.ts

```diff
@@ -44,53 +52,65 @@ export class OpendistroSecurityPlugin
   public async setup(core: CoreSetup) {
     this.logger.debug('opendistro_security: Setup');
 
-    const config$ = this.initializerContext.config.create<SecurityPluginConfigType>();
-    const config: SecurityPluginConfigType = await config$.pipe(first()).toPromise();
+    this.config$ = this.initializerContext.config.create<SecurityPluginConfigType>();
```

Comment:
what's the story behind to promote scope scope for this local variable to a class field

Reply:
I was thinking to use it in both `setup()` and `start()` function, but maybe it is better to get the `config$` from `initializerContext` in those functions separately, removed this class field